client/src/Undo/undo.hooks.ts

```diff
@@ -0,0 +1,112 @@
+import { useState, useEffect, SetStateAction, Dispatch } from 'react';
+
+import { Undo } from '../common/types';
+import { InstanceType, LimitOptions } from './undo.config';
+import * as UndoApi from './undo.api';
+import useGlobalContext from '../common/hooks/useGlobalContext';
+import { addProcess, removeProcess } from '../common/utils/globalActions';
+
+export interface TableData {
```

Comment:
In all instances we use object with this type to describe a "row of table"
Perhaps "RowData" would be more fitting name ?